utils/update_tools/update_tools.py

```diff
@@ -160,12 +165,21 @@ def main():
             branches.append(branch)
 
     try:
-        g.checkout(['-b', 'master+wip-next'])
+        g.checkout(['-b', args.branch_name])
     except Exception:
-        print("Branch master+wip-next already exists!")
-        g.checkout('master+wip-next')
+        print("Branch {} already exists!".format(args.branch_name))
+        g.checkout(args.branch_name)
 
+    # Preserving history, revert {remote}/master+wip to {remote}/master
     g.reset(['--hard', '{}/master'.format(remote)])
+    g.reset(['{}/master+wip'.format(remote)])
+    g.add(['.'])
+    g.commit(
+        "-sm \"Revert master+wip to master ({})\"".format(
+            repo.commit('{}/master'.format(remote)).hexsha
+        )
+    )
```

Comment:
I think it is worth to encapsulate this in a function as this step should be repeated in case there are merge-conflicts and a sequential merge needs to be applied.

Reply:
Sure